channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -0,0 +1,614 @@
+                let ctx = _getCoreItemCtx(coreConfig, core, extensions, pluginChain);
+                _sender.initialize(coreConfig, core, ctx, _diagLogger, _primaryChannelId, _self._unloadHooks);
+                if (_sender) {
+                    _senderInst = _sender.getXhrInst();
+                    _offlineListener.addListener((val)=> {
+                        if (!val.isOnline) {
+                            _sendNextBatchTimer && _sendNextBatchTimer.cancel();
+                        } else {
+                            _sendNextBatchTimer && _sendNextBatchTimer.refresh()
+                        }
+
+                    });
+                   
+                    _setSendNextTimer();
+                    
+                }
+
+               
+            };
+
+            _self.processTelemetry = (evt: ITelemetryItem, itemCtx?: IProcessTelemetryContext) => {
+                try {
+                    let onlineStatus = _offlineListener.isOnline();
+                    itemCtx =  itemCtx || _self._getTelCtx(itemCtx);
+
+                    
+                    if (!!onlineStatus || !_offineSupport || !_endpoint) {
+                        // if we can't get url from online sender or core config, process next
+                        _self.processNext(evt, itemCtx);
+                        return;
+                    }
+
+                    if (_hasInitialized && !_paused ) {
+                        let shouldProcess = true;
+                        if (isFunction(_offineSupport.shouldProcess)) {
+                            shouldProcess = _offineSupport.shouldProcess(evt);
+                        }
+                        if (!shouldProcess) {
+                            _self.processNext(evt, itemCtx);
+                            return;
+                        }
+                       
+                        let item = evt as IPostTransmissionTelemetryItem;
+                      
+                        //TODO: add function to better get level
+                        item.persistence = item.persistence  || (item.baseData && item.baseData.persistence) || EventPersistence.Normal; // in case the level is in baseData
+                        if (_shouldCacheEvent(_urlCfg, item) && _inMemoBatch) {
+                            if (_overrideIkey) {
+                                item.iKey = _overrideIkey;
+                            }
+                            let added = _inMemoBatch.addEvent(evt);
+                            // inMemo is full
+                            if (!added) {
+                                _flushInMemoItems();
+                                // if event can't be added again, simply drop it
+                                // TODO: better handle clean in memo events
+                                // TODO: add logs/notification
+                                _inMemoBatch.addEvent(evt);
+                                _inMemoFlushTimer && _inMemoFlushTimer.refresh();
```

Comment:
If we are always calling the `_setupInMemoTimer` (where it "resets" if not already running -- based on my comment I added to it), then I don't think this would be needed.